api/tests/acceptance/application/users/reassign-authentication-methods-route-put_test.js

```diff
@@ -0,0 +1,86 @@
+const {
+  databaseBuilder,
+  expect,
+  generateValidRequestAuthorizationHeader,
+  insertUserWithRolePixMaster,
+} = require('../../../test-helper');
+const createServer = require('../../../../server');
+const AuthenticationMethod = require('../../../../lib/domain/models/AuthenticationMethod');
+
+describe('Acceptance | Route | Users', function () {
+  describe('PUT /api/admin/users/{userId}/authentication-methods/{authenticationMethodId}/reassign', function () {
+    let server;
+    let pixMaster;
+
+    beforeEach(async function () {
+      server = await createServer();
+      pixMaster = await insertUserWithRolePixMaster();
+    });
+
+    it('should return 204 HTTP status code', async function () {
+      // given
+      const originUserId = databaseBuilder.factory.buildUser().id;
+      const targetUserId = databaseBuilder.factory.buildUser().id;
+      const authenticationMethodId = databaseBuilder.factory.buildAuthenticationMethod.withGarAsIdentityProvider({
+        userId: originUserId,
+      }).id;
+      await databaseBuilder.commit();
+
+      // when
+      const response = await server.inject({
+        method: 'PUT',
+        url: `/api/admin/users/${originUserId}/authentication-methods/${authenticationMethodId}/reassign`,
+        headers: {
+          authorization: generateValidRequestAuthorizationHeader(pixMaster.id),
+        },
+        payload: {
+          data: {
+            attributes: {
+              'user-id': targetUserId,
+              'identity-provider': AuthenticationMethod.identityProviders.GAR,
+            },
+          },
+        },
+      });
+
+      // then
+      expect(response.statusCode).to.equal(204);
+    });
+
+    it('should return 422 HTTP status code', async function () {
+      // given
+      const originUserId = databaseBuilder.factory.buildUser().id;
+      const targetUserId = databaseBuilder.factory.buildUser().id;
+      const authenticationMethodId = databaseBuilder.factory.buildAuthenticationMethod.withGarAsIdentityProvider({
+        userId: originUserId,
+      }).id;
+
+      databaseBuilder.factory.buildAuthenticationMethod.withGarAsIdentityProvider({
+        userId: targetUserId,
+        externalIdentifier: 'externalId2',
+      });
+
+      await databaseBuilder.commit();
+
+      // when
+      const response = await server.inject({
+        method: 'PUT',
+        url: `/api/admin/users/${originUserId}/authentication-methods/${authenticationMethodId}/reassign`,
+        headers: {
+          authorization: generateValidRequestAuthorizationHeader(pixMaster.id),
+        },
+        payload: {
+          data: {
+            attributes: {
+              'user-id': targetUserId,
+              'identity-provider': AuthenticationMethod.identityProviders.GAR,
+            },
+          },
+        },
+      });
+
+      // then
+      expect(response.statusCode).to.equal(422);
```

Comment:
ajouter le message d'erreur et modifier l'assertion 'when user try to reassign method to an user with GAR authentication method'

Reply:
il reste un todo ici du coup ?

Reply:
done